packages/utils/test-tools/src/enzymeTests.test.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { checkRenderConsistency, checkReRender } from './enzymeTests';
+
+const fixedStyle = {
+  backgroundColor: 'blue',
+  color: 'red',
+};
+
+const subStyle = {
+  color: 'blue',
+};
+
+const Simple = (props) => {
+  const { children, ...rest } = props;
+  return (
+    <span {...rest} style={fixedStyle}>
+      {children}
+    </span>
+  );
+};
+
+const MultiLevel = (props) => {
+  const { children, style, color, ...rest } = props;
+  return (
+    <div {...rest} style={style}>
+      <span style={subStyle}>{children}</span>
+      <span>{color}</span>
+    </div>
+  );
+};
+
+const MultiLevelBroken = (props) => {
+  const { children, style, color, ...rest } = props;
+  return (
+    <div {...rest} style={style}>
+      <span style={subStyle}>{children}</span>
+      <span style={{ color: 'green' }}>{color}</span>
+    </div>
+  );
+};
+
+let renderCount = 0;
+
+const SimpleWithHook = (props) => {
+  const { children, ...rest } = props;
+  const onKeyUp = React.useMemo(() => {
+    console.log('something');
+  }, []);
+  renderCount++;
+  return (
+    <span {...rest} {...{ onKeyUp }} style={fixedStyle}>
+      {children}
+    </span>
+  );
+};
+
+describe('enzyme component test validation', () => {
+  it('renders the simple control twice', () => {
+    checkRenderConsistency(() => <Simple>Hello</Simple>);
+  });
+
+  it('recurses into multi-level control correctly', () => {
+    checkRenderConsistency(() => <MultiLevel>World</MultiLevel>, 2);
+  });
+
+  it('handles memoed functions', () => {
```

Comment:
I really like the idea behind these test helpers!  Looking forward to seeing them get more use on our actual components.